controllers/argocdagent/create-agent-config.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# Copyright 2025 ArgoCD Operator Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+RECREATE="$1"
+
+export ARGOCD_AGENT_PRINCIPAL_NAMESPACE=argocd
+
+IPADDR=$(ip r show default | sed -e 's,.*\ src\ ,,' | sed -e 's,\ metric.*$,,')
+
+if ! test -x argocd-agentctl; then
+	echo "Please build argocd-agentctl first by running 'make cli'" >&2
+	exit 1
+fi
+
+echo "[*] Initializing PKI"
+if ! argocd-agentctl pki inspect >/dev/null 2>&1; then
+	argocd-agentctl pki init
+	echo "  -> PKI initialized."
+else
+	echo "  -> Reusing existing agent PKI."
+fi
+
+echo "[*] Creating principal TLS configuration"
+argocd-agentctl pki issue principal --upsert \
+	--principal-namespace argocd \
+	--ip "127.0.0.1,${IPADDR}"
+echo "  -> Principal TLS config created."
+
+echo "[*] Creating resouce proxy TLS configuration"
+argocd-agentctl pki issue resource-proxy --upsert \
+	--principal-namespace argocd \
+	--ip "127.0.0.1,${IPADDR}"
+echo "  -> Resource proxy TLS config created."
+
+AGENTS="agent-managed agent-autonomous"
+for agent in ${AGENTS}; do
+	echo "[*] Creating configuration for agent ${agent}"
+	if test "$RECREATE" = "--recreate"; then
+		echo "  -> Deleting existing cluster secret, if it exists"
+		kubectl -n ${ARGOCD_AGENT_PRINCIPAL_NAMESPACE} delete --ignore-not-found secret cluster-${agent}
+	fi
+	if ! argocd-agentctl agent inspect ${agent} >/dev/null 2>&1; then
+		echo "  -> Creating cluster secret for agent configuration"
+		argocd-agentctl agent create ${agent} \
+			--resource-proxy-username ${agent} \
+			--resource-proxy-password ${agent} \
+			--resource-proxy-server ${IPADDR}:9090
```

Comment:
If I'm not mistaken, when principal is running on the cluster, the `IPADDR` value should be the name of the principal Service e.g. `( crName )-agent-principal`.

Reply:
I've reviewed the code, and I believe this comment is still correct: we shouldn't use IPADDR here, but rather resource proxy Service name. 

This allows the principal's Argo CD to connect to the principal's resource proxy port